external/esp_idf_port/esp32/esp32_queue_api.c

```diff
@@ -0,0 +1,434 @@
+#include "esp_attr.h"
+#include "esp32_queue_api.h"
+#include "mq_tryreceive.h"
+/************************************************************************
+ * Pre-processor Definitions
+ ************************************************************************/
+#define MAX_QUEUE_INFO 10
+
+/************************************************************************
+ * Private Type Declarations
+ ************************************************************************/
+enum {
+	NORMAL = 0,
+	MIDDLE,
+	HIGH,
+} queue_prio_e;
+
+/************************************************************************
+ * Public Variables
+ ************************************************************************/
+const char *mq_name = "mq_wifi";
+static queue_info_t queues_info[MAX_QUEUE_INFO];
+
+/************************************************************************
+ * Public Functions
+ ************************************************************************/
+/****************************************************************************
+ * Name: queue_create_wrapper
+ *
+ * Description:
+ *	This function is the wrapper of mq_open to establish a connection between a named
+ * message queue and the calling task.
+ *
+ * Inputs:
+ *	queue_len - The length of the queue.
+ *	item_size - The size of each node of the queue.
+ *
+ * Return:
+ *	Returns the pionter of the created mqueue, but NULL in a failure.
+ *
+ ****************************************************************************/
+void *IRAM_ATTR queue_create_wrapper(uint32_t queue_len, uint32_t item_size)
+{
+#define NAME_LEN 20
+	char name[NAME_LEN];
+	bool flag = false;
+	uint32_t mq_id = 0;
+
+	for (int i = 0; i < MAX_QUEUE_INFO; i++) {
+		if (!queues_info[i].valid) {
+			flag = true;
+			mq_id = i;
+			break;
+		}
+	}
+	if (!flag) {
+		return NULL;
+	}
+
+	struct mq_attr attr;
+	attr.mq_maxmsg = queue_len;
+	attr.mq_msgsize = item_size;
+	attr.mq_flags = 0;
+	sprintf(name, "%s%d", mq_name, mq_id);
+	/*Invalid param */
+	queues_info[mq_id].mqd_fd_send = mq_open(name, O_RDWR | O_CREAT, 0666, &attr);
+	if (queues_info[mq_id].mqd_fd_send == (mqd_t) ERROR) {
+		return NULL;
+	}
+	queues_info[mq_id].mqd_fd_recv = mq_open(name, O_RDWR | O_CREAT, 0666, &attr);
+	if (queues_info[mq_id].mqd_fd_recv == (mqd_t) ERROR) {
+		return NULL;
+	}
+	queues_info[mq_id].valid = true;
+	queues_info[mq_id].mq_item_size = item_size;
+	return &queues_info[mq_id];
+}
+
+/****************************************************************************
+ * Name: queue_delete_wrapper
+ *
+ * Description:
+ *	This function is the wrapper of mq_close to close a connection between a named
+ * message queue and the calling task.
+ *
+ * Inputs:
+ *	queue - The queue to be closed.
+ *
+ * Return:
+ *
+ ****************************************************************************/
+void IRAM_ATTR queue_delete_wrapper(void *queue)
+{
+	queue_info_t *queue_info = NULL;
+
+	if (queue) {
+		queue_info = (queue_info_t *) queue;
```

Comment:
Please remove space between `(queue_info_t *)` and `queue`. 
It's the coding rule HQ requested.

Reply:
agree, done